### ISSUE TITLE
Build Issue - `yarn install` issue

### ISSUE THREAD

**Issue by curiousYi** [datalayer/jupyter-ui#101]
👋🏻 Hi!  I've been trying to get this set-up locally to poke around and have been unable to `yarn install`. I've been using the `dev-container` method recommended in the setup guide.

At first I thought maybe it was due to a lack of resources so I killed all the docker processes that I was running, but still ran into the issue. Let me know if you need more info, I'm also pretty new to using VS Code dev containers.

## Description

`yarn install` process gets killed when running from a `dev container` in VS Code.

## Reproduce

1. `git clone` repo
2. Open VSCode and run it from the container
3. In the terminal run `yarn install`

## Expected behavior
Should just be able to yarn install

![Screenshot 2023-09-24 at 10 41 27 AM](https://github.com/datalayer/jupyter-ui/assets/19179930/54d258d5-79ee-4639-acaa-1a7bedea24cd)

## Context

- Datalayer version: Commit: 6b22b7c
- Operating System and version: OS X 13.4.1 (c) (but running it from a dev container)

**Comment by echarles** [comment]
Could you give more memory to the docker image, for now it is set to 2G in the devcontainer file https://github.com/datalayer/jupyter-ui/blob/efd1afde5d13e202671d84f6e54737656fd6a1dd/.devcontainer/devcontainer.json#L6

**Comment by curiousYi** [comment]
Thank @echarles! Increasing the resources (ended up increasing it to 8 GB and 4 Cpus) got me passed the `install` step 🎉.

After successfully running `yarn build`, I'm now stuck at the `yarn start` step. Was wondering if there's anything obvious I'm missing? 

<img width="1114" alt="Screenshot 2023-09-27 at 8 57 25 AM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/ab9b09e8-b745-4280-983d-53827ee3b160">

**Comment by curiousYi** [comment]
I also opened a PR to update the documentation so other users who run into the same issue could help unstuck themselves.

https://github.com/datalayer/jupyter-ui/pull/107

**Comment by echarles** [comment]
> Thank @echarles! Increasing the resources (ended up increasing it to 8 GB and 4 Cpus) got me passed the install step 🎉.

Thos values are really freaking... I don't use devecontainers, but when I have setup that, the 2gb were OK for me. On which OS are you running?

> After successfully running yarn build, I'm now stuck at the yarn start step. Was wondering if there's anything obvious I'm missing?

The logs seems to show that the server starts on port 3208, but after crashes before the bundle is created... The log says to add `--verbose`, have you tried that? I can try on my laptop to see how it goes if you don't get find a solution soon.

**Comment by curiousYi** [comment]
> > Thank @echarles! Increasing the resources (ended up increasing it to 8 GB and 4 Cpus) got me passed the install step 🎉.
> 
> Thos values are really freaking... I don't use devecontainers, but when I have setup that, the 2gb were OK for me. On which OS are you running?

I'm running OS Ventura. I'm not sure the values needed to be that high but just noting that's the change I made and it unblocked me. 

> 
> > After successfully running yarn build, I'm now stuck at the yarn start step. Was wondering if there's anything obvious I'm missing?
> 
> The logs seems to show that the server starts on port 3208, but after crashes before the bundle is created... The log says to add `--verbose`, have you tried that? I can try on my laptop to see how it goes if you don't get find a solution soon.

I did try that. Now I ran `yarn start` but am getting a different set of errors so that's...progress? For some reason it cannot find the modules despite me running `yarn install` in the react repo and seeing the dependencies listed. I'll try running it with `--stats-error-details` later when I have time.


<img width="1133" alt="Screenshot 2023-09-27 at 10 39 09 AM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/caa45f23-7012-4fd9-9c66-74084847963d">

<img width="1160" alt="Screenshot 2023-09-27 at 10 39 15 AM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/e196d35f-dee7-4e73-b45b-01e5ba02cac7">


<img width="1123" alt="Screenshot 2023-09-27 at 10 38 47 AM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/f832b8bd-0c2b-4f67-bc3f-157f1b2405f5">

<img width="1139" alt="Screenshot 2023-09-27 at 10 38 59 AM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/0e5f4f86-9624-4d48-9dd3-6857e201ee3e">

**Comment by echarles** [comment]
Are you sure yarn build has run successfully?

**Comment by curiousYi** [comment]
> Are you sure yarn build has run successfully?

`yarn build` did NOT run successfully actually. See output below.
I tried changing into the individual packages that had build issues and running the build command there. But still blocked at the build step.

![Screenshot 2023-09-27 at 7 30 47 PM](https://github.com/datalayer/jupyter-ui/assets/19179930/d25ebc56-6931-4194-8c57-6597148701ed)

**Comment by echarles** [comment]
main branch has been unstable those last days. The latest commit builds fine on macos and linux (windows is failing since some time, need to look at that...), see https://github.com/datalayer/jupyter-ui/actions/runs/6334472183

Could you pull the latest commits and try again (maybe start from a completely blank repo, so remove and clone, to ensure you don't have remaining node_modules or tsconfig.tsbuildinfo files).

**Comment by curiousYi** [comment]
Hey @echarles, I followed your instructions (pulled a completely blank repo with the latest commits) but ran into the following install issue.

<img width="1317" alt="Screenshot 2023-10-06 at 3 12 32 PM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/046d5bb5-54d1-4cc5-a795-129159e4cbfd">

<img width="1304" alt="Screenshot 2023-10-06 at 3 12 44 PM" src="https://github.com/datalayer/jupyter-ui/assets/19179930/62f09856-398a-4abb-aea6-15dd7f443512">

**Comment by echarles** [comment]
In the meantime other users have reported successful installation in devcontainers. I will close it for now. Please reopen or open another issue if you are still facing troubles. All the best!
